Point cloud shader

Vesper-Works opened the pull request.

Vesper-Works committed:
Update .gitignore

Vesper-Works committed:
Prototype complete

Press K to cast a bajillion rays randomly in a sphere

Vesper-Works committed:
Materials Updated, Points have lifetime

HFinch13 committed:
Unity Push 1

Test of pushing Unity Assests to the repo

Vesper-Works committed:
Colour cached

On scene change SoundUtil will cache all the material colours in a dictionary with the transforms of that object. If any object is instantiated, they will not have a colour.

Vesper-Works committed:
Merge branch 'Level_HF' into point_cloud_shader

HFinch13 committed:
Commit 2; test to see if it works

HFinch13 committed:
Commit

Coloured Level geometry and fixed seam issues as well as created next level sections. the magnetisation keeps changing which is annoying

Vesper-Works committed:
Revert "Colour cached"

This reverts commit f64a820fc3789ecceeedafb59f4441bf70fcce29.

Vesper-Works committed:
Revert "Revert "Colour cached""

This reverts commit 4bbdd019b657617a34250cf40686f320fb421a86.

Vesper-Works committed:
Revert "Merge branch 'Level_HF' into point_cloud_shader"

This reverts commit d208cf48dc170ce7cc77662669c3aa53a5235fdc, reversing
changes made to f64a820fc3789ecceeedafb59f4441bf70fcce29.

Vesper-Works committed:
Sound maker

Vesper-Works committed:
Merge branch 'Level_HF' into point_cloud_shader

Vesper-Works committed:
Updated materials, added gizmos

Vesper-Works committed:
Finished up - one problem left to fix

Only problem here is some surfaces will intercept the raycast, but not create a point, and I have no idea why.

Vesper-Works committed:
Added invert option

Vesper-Works committed:
Changed to my own Point Cloud Testing scene

Vesper-Works committed:
Merge remote-tracking branch 'origin/player_character' into point_cloud_shader

StevieGoodman committed:
Created test assemblies

StevieGoodman committed:
Removed prototyping demo scene

Signed-off-by: Stevie <mail@steviegoodman.com>

StevieGoodman committed:
Added empty play test

Vesper-Works committed:
Odin

Vesper-Works committed:
Merge branch 'point_cloud_shader' of https://github.com/CMP2804/Assessment3 into point_cloud_shader

Vesper-Works committed:
Points are no longer generated behind walls

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+        }
+        private void GetNormalsDataFromMesh()
+        {
+            foreach (Mesh mesh in _meshes)
```
This `foreach` loop doesn't need braces - it has only 1 line

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+
+            _normalBuffer = new ComputeBuffer(_normals.Count, 3 * sizeof(float));
+            _normalBuffer.SetData(_normals);
+            //computeShader.SetBuffer(0, "_Points", pointBuffer);
```
This comment should be removed.

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+
+            _pointBuffer = new ComputeBuffer(_vertices.Count, 3 * sizeof(float));
+            _pointBuffer.SetData(_vertices);
+            //computeShader.SetBuffer(0, "_Points", pointBuffer);
```
This comment should be removed.

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+
+        private void GetPositionsDataFromMesh()
+        {
+            foreach (Mesh mesh in _meshes)
```
This `foreach` loop doesn't need braces - it has only 1 line

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+
+        private void GetEachMeshRecursively(Transform parent)
+        {
+            if (parent.TryGetComponent(out MeshFilter meshFilter))
```
Remove braces

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+            pointMaterial.SetFloat("_intensity", 1);
+            pointMaterial.SetVector("_worldPos", this.transform.position);
+            pointMaterial.SetVector("_color", new Vector4(1, 1, 1, 1));
+            //material.SetFloat("_ColorFromTextureLerp", colorFromTextureLerp);
```
These comments should be removed

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+            //material.SetFloat("_ColorFromTextureLerp", colorFromTextureLerp);
+            //material.SetFloat("_UseAlpha", useAlpha?1:0);
+            pointMaterial.SetMatrix("_quaternion",
+                Matrix4x4.TRS(new Vector3(0, 0, 0), transform.rotation, new Vector3(1, 1, 1)));
```
Inconsistent indentation. To provide context, I would extract variables for the first and last arguments in `Matrix4x4.RTS()`, as well as the function return itself.

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+    {
+        [SerializeField] private Material pointMaterial = default;
+        [SerializeField] private Mesh pointMesh = default;
+        [SerializeField] private ComputeShader computeShader;
```
For consistency (and functionality), replace all `[SerializeField]` with `[OdinSerialize]` as this supports more advanced types

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudObject.cs:
```diff
@@ -0,0 +1,85 @@
+
+namespace cmp2804.Point_Cloud
+{
+    public class PointCloudObject : MonoBehaviour
```
Derive from `SerializedMonoBehaviour` instead for access to property serialization

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudRenderer.cs:
```diff
@@ -0,0 +1,178 @@
+        }
+        private void Update()
+        {
+            // for (int i = 0; i < _lifespans.Count; i++)
```
Pleass remove comments

kit-calloway commented on Assets/Scripts/Point Cloud/PointCloudRenderer.cs:
```diff
@@ -0,0 +1,178 @@
+            bool reset = false;
+            for (int i = 0; i < _lifespans.Count; i++)
+            {
+                if (_lifespans[i] <= 0)
```
I would recommend extracting the contents of this `if` statement and replacing it with `if (_lifespans[i] <= 0) functionName();`

kit-calloway commented on Assets/Scripts/Point Cloud/SoundMaker.cs:
```diff
@@ -0,0 +1,88 @@
+{
+    public class SoundMaker : MonoBehaviour
+    {
+        [Header("Emission settings")] [SerializeField]
```
Recommend replacing this with `[Title(title, description)` as this is Odin's implementation

StevieGoodman reviewed (changes requested):
Okay, there's nothing game-breaking here, but I've given a few tips that I would recommend acting upon, but don't feel compelled to

Vesper-Works committed:
Fixed formatting

Using Rider's clean function on the project, that's why weird files have been changed too, I could fix it but cba and it doesn't matter

StevieGoodman reviewed (changes requested):
Please resolve merge conflicts

Vesper-Works committed:
Merge branch 'submission' into point_cloud_shader

Signed-off-by: Vesper <59376295+Vesper-Works@users.noreply.github.com>

Vesper-Works commented:
Ready

Vesper-Works committed:
Fixed normals issue and Odin serialisation

StevieGoodman reviewed (approved):
https://media.istockphoto.com/id/513921039/photo/illustration-of-a-oool-yellow-smiley-with-sunglasses.jpg?s=612x612&w=0&k=20&c=hhVQxXTUhmcZLv2QrZ2WE2p7inzxQIA5H6XP8jPrQXw=
